pkg/network/node/networkpolicy.go

```diff
@@ -125,13 +127,28 @@ func (np *networkPolicyPlugin) Start(node *OsdnNode) error {
 	}
 
 	otx := node.oc.NewTransaction()
-	for _, cn := range np.node.networkInfo.ClusterNetworks {
-		// Must pass packets through CT NAT to ensure NAT state is handled
-		// correctly by OVS when NAT-ed packets have tuple collisions.
-		// https://bugzilla.redhat.com/show_bug.cgi?id=1910378
-		otx.AddFlow("table=21, priority=200, ip, nw_dst=%s, ct_state=-rpl, actions=ct(commit,nat(src=0.0.0.0),table=30)", cn.ClusterCIDR.String())
+
+	// Register all pod-network-internal connections with conntrack, so we can later
+	// allow replies to them to skip policy enforcement. (We don't do this for traffic
+	// leaving the pod network, because those packets usually need to get processed by
+	// iptables, but the kernel won't do that if we have already called ct(commit) on
+	// them.)
+	//
+	// The dummy nat(src=0.0.0.0) ensures we handle tuple collisions; see
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1910378
+	for _, scn := range np.node.networkInfo.ClusterNetworks {
+		for _, dcn := range np.node.networkInfo.ClusterNetworks {
+			otx.AddFlow("table=30, priority=100, ip, nw_src=%s, nw_dst=%s, ct_state=-rpl, actions=ct(commit,table=31)", scn.ClusterCIDR.String(), dcn.ClusterCIDR.String())
+		}
 	}
+
+	// skip policy enforcement for egress when the destination address is a ServiceIP
+	otx.AddFlow("table=27, priority=300, ip, nw_dst=%s , actions=goto_table:30", node.networkInfo.ServiceNetwork.String())
```

Comment:
We're not _skipping_ it, we're just postponing it. Maybe:

    // Egress enforcement for pod-to-Service IP will happen after the packets are rewritten by
    // iptables and come back via table 25.

Also, I generally like to add the rules in the "right" order, so that the order you read them in in the code is the same as the order they get processed in... oh, hah, but it was already wrong. OK, can you fix it so this goes:

- `otx := node.oc.NewTransaction()`
- "service IP" comment plus the priority 300 rule for service IPs
- "replies" comment plus the priority 200 rules
- the big "conntrack" comment plus the loop with the priority 100 rules

Reply:
done